src/main/webapp/scripts/cards.js

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Function that retrive vendors given query parameters
+ * TODO: Add fetch from servelt & retrieve picture from blobstore
+ */
+querySalecards = () => {
+    // Once cards are retrieved...
+    const salecardTemplate = document.getElementById('salecard-template');
+    const salecardsContainer = document.getElementById('salecards-container');
+
+    //Clean previously retrieved cards
+    salecardsContainer.innerHTML = '';
+
+    Object.keys(vendors).forEach(vendorNumber => {
+        let vendor = vendors[vendorNumber];
+        let salecard = vendor.saleCard;
+        let salecardCloned = salecardTemplate.content.cloneNode(true);
+        salecardCloned.getElementById('business-name').innerHTML = salecard.businessName;
+        salecardCloned.getElementById('business-description').innerHTML = salecard.description;
+        salecardCloned.getElementById('vendor-name').innerHTML = `${vendor.firstName} ${vendor.lastName}`;
+        salecardCloned.getElementById('vendor-phone').innerHTML = vendor.phoneNumber;
+        salecardCloned.getElementById('vendor-distance').innerHTML = `${salecard.location.distance}m`;
```

Comment:
I think we shouldn't use `innerHTML` to set the content, because of cross-site scripting attacks. Instead use: `textContent`. [This is what I found about it](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML#Security_considerations)

Reply:
Great advice, thank you!